src/modules/order/conteiners/PaymentDetail/index.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import styles from './index.module.scss';
+import { useTranslation } from 'react-i18next';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import clsx from 'clsx';
+import PaymentInput from 'modules/order/components/PaymentInput';
+
+export interface paymentForm {
+  cardNumber: string;
+  cardHolder: string;
+  month: number;
+  year: number;
+  cvv: number;
+}
+
+const normalizeCardNumber = (value: string) => {
+  return (
+    value
+      ?.replace(/\D/g, '')
+      .match(/.{1,4}/g)
+      ?.join(' ')
+      .slice(0, 19) || ''
+  );
+};
+
+const normalizeCVV = (value: string) => {
+  return value?.replace(/\D/g, '').slice(0, 3) || '';
+};
+
+const PaymentDetail = () => {
+  const { t: validationT } = useTranslation('validation');
+  const { t: defaultT } = useTranslation();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<paymentForm>({ mode: 'onBlur' });
+
+  const onSubmit: SubmitHandler<paymentForm> = () => {
+    // axios request to backend payment endpoint
+  };
+
+  const requiredInput = {
+    required: {
+      value: true,
+      message: validationT('required'),
+    },
+  };
+
+  const cardNumberRules = {
+    ...requiredInput,
+    pattern: {
+      value: /(?:\d[ ]*?){16}/,
+      message: validationT('invalidCardNumber'),
+    },
+  };
+
+  const cvvRules = {
+    ...requiredInput,
+    pattern: {
+      value: /\d{3}/,
+      message: validationT('invalidCardNumber'),
+    },
+  };
```

Comment:
we could create util function or hook that gets pattern and error message

Reply:
Should we use it as global hook?

Reply:
Example: 
```
import { useTranslation } from 'react-i18next';

interface UseValidationRulesProps{
  value: string;
  message: string;
  required: boolean;
}

export const useValidationRules = ({ value, message, required}: UseValidationRulesProps) => {
  const { t: validationT } = useTranslation('validation');
  
  const requiredInput = required ? {
    required: {
      value: true,
      message: validationT('required'),
    },
  } : {};

  const rules = {
    ...requiredInput,
    pattern: {
      value: value,
      message: message,
    }
  };

  return rules;
};

```